src/puppetlabs/trapperkeeper/services/webserver/jetty9_config.clj

```diff
@@ -52,23 +52,22 @@
 ;;; their implicit defaults reflect the security issue.  The latter is far more
 ;;; risky for our downstream apps, thus it was decided that it makes sense to
 ;;; keep these overrides.
-;;;
-;;; Also note that w/rt the default list of acceptable ciphers, we're deliberately
-;;; excluding all diffie-helman ciphers due to some old JDK bugs:
-;;;
-;;; http://bugs.java.com/bugdatabase/view_bug.do?bug_id=8014618
-;;; https://github.com/puppetlabs/puppetdb/commit/03e020dc85b83d6c83c9992ca6bd14f57e8fc91a
-;;;
-;;; These have been fixed in recent versions of the JDK, and it would be nice
-;;; to be able to add the DH ciphers back in at some point, but we can't do that
-;;; until we're certain that our minimum supported JDK versions for all of our
-;;; supported distros will contain the relevant fixes.
-;;;
 (def acceptable-ciphers
-  ["TLS_RSA_WITH_AES_256_CBC_SHA256"
+  ["TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA256"
+   "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA"
+   "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256"
+   "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"
```

Comment:
I don't believe we have any code in our Trapperkeeper-based apps which configure Bouncycastle as a JSSE provider, so I believe we should be looking at this in terms of what the default SunJSSE provider in the JDK would use.  We're only using Bouncycastle for its PKIX API - helper methods behind the Puppet Server Certificate Authority service in the [jvm-ssl-utils](https://github.com/puppetlabs/jvm-ssl-utils) project.

It appears that [the SupportedEllipticCurvesExtension class](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/sun/security/ssl/SupportedEllipticCurvesExtension.java) in the JDK source shows the available curves - P-256 (secp256r1) being preferred - but I'm not finding any references to how these could be user / programmatically configured outside of the JDK.  Is it a problem if we can't find / document a way for users to configure which curves are used?